pilot/pkg/model/push_context.go

```diff
@@ -1421,7 +1425,11 @@ func (ps *PushContext) initDestinationRules(env *Environment) error {
 	destRules := make([]Config, len(configs))
 
 	for i := range destRules {
-		destRules[i] = configs[i].DeepCopy()
+		var tmpCopy networking.DestinationRule
```

Comment:
Is it possible to change the `DeepCopy` implementation to have this logic?

Reply:
Yes we can. But i think it is a little tough, have to get the `ConfigMeta.Type`, `ConfigMeta.Group` and `ConfigMeta.Version`, then we need to do each kind of deep copy separately

Reply:
I meant some thing similar - not tried though - but rough sketch, would it work?

```
func (c Config) DeepCopy() Config {
	var clone Config
	cs := proto.Clone(c.Spec)
	clone.ConfigMeta = c.ConfigMeta
	clone.Spec = cs
	return clone
}
```

Reply:
Ah, good to know that the DeepCopyInto is implemented by proto.Clone